### DISCUSSION TITLE
Rename `send` to `createAction` ?

### DISCUSSION THREAD

**Discussion by redbar0n** [statelyai/xstate#3597] [Ideas] [answered]
Reading through the docs, seeing such things as big red and yellow warnings:

> All action creators documented here return action objects; it is a pure function that only returns an action object and **does not imperatively send an event.** Do not imperatively call action creators; they will do nothing!

https://xstate.js.org/docs/guides/actions.html#action-order

> The send(...) function is an **action creator**; it is a pure function that only returns an action object and **does not imperatively send an event**.

https://xstate.js.org/docs/guides/actions.html#send-action

> Again, the send(...) function is an action creator and **will not imperatively send an event**. Instead, it returns an **action object** that describes where the event will be sent to:

https://xstate.js.org/docs/guides/actions.html#send-action

Documentation is good. But self-documenting names and no needed documentation is even better.

These warnings indicate to me that `send` perhaps should have been named `createAction` instead?

**Comment by Andarist** [comment, marked as answer]
`send` is just one of many action creators - so a `createAction` name doesn't fit this. Ofc, we could have had `createAction('send', {})` API but that would get quite verbose very quickly.

**Comment by davidkpiano** [comment]
Note that `send` will be renamed to `sendTo` in v5 to avoid confusion, and to make a simpler API.
